linter: `DummyRuleMap` index signature causes TS2411 errors in `oxlint/dist/index.d.ts`

> **Note:** This bug report was AI-generated (but human-reviewed before posting).

## Version

- `oxlint`: v1.65.0 and v1.66.0 (confirmed present in both)
- `typescript`: v6.0.3
- `tsgo` (`@typescript/native-preview`): v7.0.0-dev.20260512.1

## Description

Using `defineConfig` from `oxlint` in a TypeScript config file causes hundreds of `TS2411` type errors originating from `oxlint/dist/index.d.ts`, not from user code. The errors come from a mismatch between the optional named properties and the catch-all index signature in the `DummyRuleMap` interface.

## Root cause

In `oxlint/dist/index.d.ts`, the `DummyRuleMap` interface declares all named rule properties as optional:

```ts
interface DummyRuleMap {
  "accessor-pairs"?: DummyRule;   // type is DummyRule | undefined
  "array-callback-return"?: DummyRule;
  // ... ~800 more optional properties ...
  yoda?: DummyRule;
  [k: string]: DummyRule;         // index signature does NOT allow undefined
}
```

TypeScript's TS2411 rule requires that all named property types be assignable to the index signature type. Because the named properties are optional (`?:`), their type is `DummyRule | undefined`, which is not assignable to the index signature type `DummyRule`.

**Fix:** change the index signature to allow `undefined`:

```ts
- [k: string]: DummyRule;
+ [k: string]: DummyRule | undefined;
```

## Steps to reproduce

1. Create a project with `oxlint` installed.
2. Create an `oxlint.config.ts` using `defineConfig`:

```ts
import { defineConfig } from 'oxlint';

export default defineConfig({
  rules: {
    'no-console': 'error',
  },
});
```

3. Run `tsc --noEmit` (or `tsgo`).

## Expected behavior

No type errors — `oxlint`'s own type definitions should be internally consistent.

## Actual behavior

Hundreds of TS2411 errors emitted from `node_modules/oxlint/dist/index.d.ts`:

```
node_modules/oxlint/dist/index.d.ts(484,3): error TS2411: Property '"accessor-pairs"' of type 'DummyRule | undefined' is not assignable to 'string' index type 'DummyRule'.
node_modules/oxlint/dist/index.d.ts(485,3): error TS2411: Property '"array-callback-return"' of type 'DummyRule | undefined' is not assignable to 'string' index type 'DummyRule'.
... (one error per named property in DummyRuleMap)
```

--- a/apps/oxlint/scripts/generate-config-types.ts
+++ b/apps/oxlint/scripts/generate-config-types.ts
@@ -70,6 +70,18 @@ function sanitizeSchema(value: any): void {
 
 sanitizeSchema(schema);
 
+const dummyRuleMap = schema.definitions?.DummyRuleMap;
+const dummyRuleMapAdditionalProperties = dummyRuleMap?.additionalProperties;
+if (
+  typeof dummyRuleMapAdditionalProperties !== "object" ||
+  dummyRuleMapAdditionalProperties === null
+) {
+  throw new Error("Expected DummyRuleMap.additionalProperties in the oxlint config schema.");
+}
+// Named rule properties are optional, so the string index signature must
+// accept `undefined` to keep the generated declaration internally valid.
+dummyRuleMapAdditionalProperties.tsType = "DummyRule | undefined";
+
 const bannerComment =
   "/*\n" +
   " * This file is generated from npm/oxlint/configuration_schema.json.\n" +

--- a/apps/oxlint/src-js/package/config.generated.ts
+++ b/apps/oxlint/src-js/package/config.generated.ts
@@ -1321,7 +1321,7 @@ export interface DummyRuleMap {
   "vue/valid-define-props"?: DummyRule;
   "vue/valid-next-tick"?: DummyRule;
   yoda?: DummyRule;
-  [k: string]: DummyRule;
+  [k: string]: DummyRule | undefined;
 }
 /**
  * Configure the behavior of linter plugins.

